components/SearchBar.vue

```diff
@@ -2,92 +2,180 @@
+                <span class="self-center text-primary-text-inverted">{{
+                    $t("searchBar.search")
+                }}</span>
             </button>
         </div>
     </div>
 </template>
 
 <script setup lang="ts">
-import { ref, Ref, watchEffect } from 'vue'
-import { useSearchResultsStore } from '~/stores/searchResultsStore.js'
-import { useLocationsStore } from '~/stores/locationsStore.js'
-import { useSpecialtiesStore, SpecialtyDisplayOption } from '~/stores/specialtiesStore.js'
-import { Locale, Specialty } from "~/typedefs/gqlTypes.js"
-import { useLocaleStore, LocaleDisplay } from '~/stores/localeStore.js'
+import { ref, Ref, watchEffect } from "vue";
+import { useSearchResultsStore } from "~/stores/searchResultsStore.js";
+import { useLocationsStore } from "~/stores/locationsStore.js";
+import {
+    useSpecialtiesStore,
+    SpecialtyDisplayOption,
+} from "~/stores/specialtiesStore.js";
+import { Locale, Specialty } from "~/typedefs/gqlTypes.js";
+import { useLocaleStore, LocaleDisplay } from "~/stores/localeStore.js";
 
-const localeStore = useLocaleStore()
-const locationsStore = useLocationsStore()
-const searchResultsStore = useSearchResultsStore()
-const specialtiesStore = useSpecialtiesStore()
+const localeStore = useLocaleStore();
+const locationsStore = useLocationsStore();
+const searchResultsStore = useSearchResultsStore();
+const specialtiesStore = useSpecialtiesStore();
 
-await locationsStore.fetchLocations()
+await locationsStore.fetchLocations();
 
-const languageOptions = [{
-    code: '',
-    simpleText: '----Any----',
-    displayText: '----Any----'
-}, ...localeStore.mvpLocaleDisplayOptions]
+const languageOptions = [
+    {
+        code: "",
+        simpleText: "----Any----",
+        displayText: "----Any----",
+    },
+    ...localeStore.mvpLocaleDisplayOptions,
+];
 
-const locationDropdownOptions: Ref<string[]> = ref(locationsStore.citiesDisplayOptions)
-const specialtyDropdownOptions: Ref<SpecialtyDisplayOption[]> = ref(specialtiesStore.specialtyDisplayOptions)
-const languageDropdownOptions: Ref<LocaleDisplay[]> = ref(languageOptions)
+const locationDropdownOptions: Ref<string[]> = ref(
+    locationsStore.citiesDisplayOptions
+);
+const specialtyDropdownOptions: Ref<SpecialtyDisplayOption[]> = ref(
+    specialtiesStore.specialtyDisplayOptions
+);
+const languageDropdownOptions: Ref<LocaleDisplay[]> = ref(languageOptions);
 
-const selectedSpecialty: Ref<Specialty | String> = ref('')
-const selectedLocation: Ref<string> = ref('')
-const selectedLanguage: Ref<Locale | String> = ref('')
+const selectedSpecialty: Ref<Specialty | String> = ref("");
+const selectedLocation: Ref<string> = ref("");
+const selectedLanguage: Ref<Locale | String> = ref("");
+
+function createCityTable() {
+    let lookupCityCount: { [key: string]: number } = {};
+    const cities = locationsStore.citiesDisplayOptions;
+
+    for (let i = 0; i < cities.length; i++) {
```

Comment:
You can use a forEach loop instead  here to make it more readable

Reply:
changed it to a` forEach` loop